### ISSUE TITLE
[FR] Easily add a search bar

### ISSUE THREAD

**Issue by rtheodoro** [hugo-apero/hugo-apero#56]
HI ! Thanks for your package, it's awesome!

I would like to put a search bar in my blog.

I found this: https://gohugo.io/tools/search/
But I was not able to implement with this because I basically only know R.

It would be nice if the package have this option.

**Comment by mccarthy-m-g** [comment]
I don’t have the JavaScript experience (or time) to implement this, but I did some searching because I’d love this feature too!

[lunr.js](https://lunrjs.com/) seems like a good option, and it’s one of the ones recommended in the Hugo docs page @rtheodoro shared above. Here is a really great Hugo specific tutorial if someone wants to give this a go:

- [Add Search to Your Static Site with Lunr.js (Hugo, Vanilla JS)](https://aaronluna.dev/blog/add-search-to-static-site-lunrjs-hugo-vanillajs/)

We might not need every feature there, but I really like how it implements the search feature like it’s a single page app (rather than redirecting to a new page when you search).

One other thing we might want to think about in implementing this is where to actually put the search box. I think a nice place for it would be in the sidebar for the various page types (e.g., blog, talks, etc.). And then whether it’s displayed or not can be toggled with a YAML parameter.

**Comment by apreshill** [comment]
I agree- I think I may be able to take a swing but I know @dcossyleon used this post to implement search in the new and improved RStudio blog, so leaving the link here in case someone wants to dig into it!

https://twitter.com/dcossyle/status/1461690409642610693?s=20

https://palant.info/2020/06/04/the-easier-way-to-use-lunr-search-with-hugo/

**Comment by rtheodoro** [comment]
Thanks for the information! 

Unfortunately, I still haven't been successful in adding the search bar :( Sounds easy with the palant tutorial, but I don't know where I'm failing.

About the place to put it, I think would be nice to put in the header too, like palant did in `MemE` theme, here https://github.com/reuixiy/hugo-theme-meme/tree/1322fdde91af3a147ce9881892a432ebfaf9b535

Again, thanks a lot!

**Comment by derrygeoff** [comment]
You may find this handy to add a [hugo contact form](https://fabform.io/a/hugo-contact-form)
